src/activity_item_handling.cpp

```diff
@@ -394,6 +395,64 @@ void activity_handlers::drop_do_turn( player_activity *act, player *p )
     put_into_vehicle_or_drop( *p, obtain_activity_items( *act, *p ), pos );
 }
 
+void activity_handlers::washing_finish( player_activity *act, player *p )
+{
+    const std::string cur_time = to_string_time_of_day( calendar::turn );
+
+    auto items = reorder_for_dropping( *p, convert_to_indexes( *act ) );
+
+    // Check again that we have enough water and soap incase the amount in our inventory changed somehow
+    // Consume the water and soap
+    int required_water = 0;
+    int required_cleanser = 0;
+
+    std::list<act_item>::iterator item_to_clean;
+    for( item_to_clean = items.begin(); item_to_clean != items.end(); ++item_to_clean ) {
```

Comment:
Why are  you declaring the loop variable outside of the loop? It's only used inside of it.

This whole loop should probably be a simple range-based loop: `for( const act_item &mod : items ) {`.